space/internal/components/aclnotifications/notifications.go

```diff
@@ -0,0 +1,229 @@
+		err := n.iterateAclContent(ctx, record, permissions, space, aclId, aclData)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (n *AclNotificationSender) iterateAclContent(ctx context.Context,
+	record *list.AclRecord,
+	permissions list.AclPermissions,
+	space clientspace.Space,
+	aclId string,
+	aclData *aclrecordproto.AclData,
+) error {
+	for _, content := range aclData.AclContent {
+		if permissions.CanManageAccounts() {
+			if reqJoin := content.GetRequestJoin(); reqJoin != nil {
+				if err := n.sendJoinRequest(ctx, reqJoin, space, record.Id, aclId); err != nil {
+					return err
+				}
+			}
+			if reqLeave := content.GetAccountRemove(); reqLeave != nil {
+				if err := n.sendAccountRemove(ctx, space, record, aclId); err != nil {
+					return err
+				}
+			}
+		}
+		if reqApprove := content.GetRequestAccept(); reqApprove != nil {
+			if err := n.sendParticipantRequestApprove(reqApprove, space, record.Id, aclId); err != nil {
+				return err
+
+			}
+		}
+	}
+	return nil
+}
+
+func (n *AclNotificationSender) sendJoinRequest(ctx context.Context,
+	reqJoin *aclrecordproto.AclAccountRequestJoin,
+	space clientspace.Space, id, aclId string,
+) error {
+	var name, iconCid string
+	pubKey, err := crypto.UnmarshalEd25519PublicKeyProto(reqJoin.InviteIdentity)
+	if err != nil {
+		return err
+	}
+	profile := n.getProfileData(ctx, pubKey.Account())
+	if profile != nil {
+		name = profile.Name
+		iconCid = profile.IconCid
+	}
+	err = n.notificationService.CreateAndSend(&model.Notification{
+		Id:      id,
+		IsLocal: false,
+		Payload: &model.NotificationPayloadOfRequestToJoin{RequestToJoin: &model.NotificationRequestToJoin{
+			SpaceId:      space.Id(),
+			Identity:     pubKey.Account(),
+			IdentityName: name,
+			IdentityIcon: iconCid,
+		}},
+		Space:     space.Id(),
+		AclHeadId: aclId,
+	})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (n *AclNotificationSender) sendParticipantRequestApprove(reqApprove *aclrecordproto.AclAccountRequestAccept,
+	space clientspace.Space,
+	id, aclId string,
+) error {
+	identity, _, _ := n.identityService.GetMyProfileDetails()
+	pubKey, err := crypto.UnmarshalEd25519PublicKeyProto(reqApprove.Identity)
+	if err != nil {
+		return err
+	}
+	account := pubKey.Account()
+	if account != identity {
+		return nil
+	}
+	err = n.notificationService.CreateAndSend(&model.Notification{
+		Id:      id,
+		IsLocal: false,
+		Payload: &model.NotificationPayloadOfParticipantRequestApproved{
+			ParticipantRequestApproved: &model.NotificationParticipantRequestApproved{
+				SpaceID:    space.Id(),
+				Permission: mapProtoPermissionToAcl(reqApprove.Permissions),
+			},
+		},
+		Space:     space.Id(),
+		AclHeadId: aclId,
+	})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (n *AclNotificationSender) getProfileData(ctx context.Context, account string) *model.IdentityProfile {
+	ctxWithTimeout, _ := context.WithTimeout(ctx, time.Second*30)
```

Comment:
I think we can't show notification without participant name, so we need to wait it until it's loaded